stackrox/stackrox
# Delete leftover watchLocation in clusterSagas

## Description

Do not make unneeded /v1/clusters requests, especially for main dashboard.

See if it solves intermittent integration test failure:

> Cluster management should indicate which clusters are managed by Helm and the Operator

> Error: The following error originated from your application code, not from Cypress. It was caused by an unhandled promise rejection.

> Network Error

* https://app.circleci.com/pipelines/github/stackrox/stackrox/10288/workflows/85211755-5702-44f5-8f5b-2a66a0bfc258/jobs/470401

### Analysis of requests

Screen shot from CI for test fail in which clusters list displays response from actual request:
![cypress-ci-clusters](https://user-images.githubusercontent.com/11862657/165314327-f0aba6fb-ae9e-40f1-83c8-d21f7dec5ad5.png)

Screen shot from local deployment for test pass in which clusters list displays fixture from mock request:
<img width="1347" alt="cypress-local-clusters" src="https://user-images.githubusercontent.com/11862657/165314833-e368e0ec-9237-488a-be1a-3fb774199d1b.png">

* /v1/clusters is the first request for /main/dashboard
    ![ui-dashboard](https://user-images.githubusercontent.com/11862657/165314401-e5ead246-a204-4288-9798-131b6ea0f9c4.png)

* /v1/clusters is not the first request for /main/clusters
    ![ui-clusters](https://user-images.githubusercontent.com/11862657/165314432-4bb40acd-3518-4b68-bc42-00f9fd778cb5.png)

Investigate logimbue-data.json file:
1. Within `clusters` property, `"byId": {}` therefore there has not yet been a successful response to a /v1/clusters request through redux sagas
2. There are 2 /v1/clusters requests, the first has 200 status and the second has 0 status:

```json
        {
          "timestamp": 1650910006.382,
          "type": "http",
          "category": "xhr",
          "data": { "method": "GET", "url": "/v1/clusters", "status_code": 200 }
        },
        {
          "timestamp": 1650910006.382,
          "category": "redux-action",
          "message": "serverError/RECORD_SUCCESS"
        },
        {
          "timestamp": 1650910006.429,
          "type": "http",
          "category": "xhr",
          "data": {
            "method": "GET",
            "url": "/v1/cluster-defaults",
            "status_code": 200
          }
        },
        {
          "timestamp": 1650910006.429,
          "category": "redux-action",
          "message": "serverError/RECORD_SUCCESS"
        },
        {
          "timestamp": 1650910006.497,
          "type": "http",
          "category": "xhr",
          "data": { "method": "GET", "url": "/v1/clusters", "status_code": 0 }
        },
        {
          "timestamp": 1650910006.498,
          "category": "redux-action",
          "message": "serverError/RECORD_ERROR"
        }
```

Is there a racing condition which causes a conflict between:
* /v1/clusters request for main dashboard (initial default route) in redux sagas
* /v1/clusters request for clusters page (visited route)
*  `intercept` with fixture in clusters integration tests?

### Analysis of reducers and sagas

1. reducers/cluster exports the following selectors:
    * getClustersById: 0 occurrences
    * getClusters: 3 occurrences in Containers/Network
    * getSelectedClusterId: 0 occurrences
    * getWizardCurrentPage: 1 occurrence in sagas/clusterSagas
    * getWizardClusterId: 2 occurrences in sagas/clusterSagas
    * search selectors: residue

2. reducers/cluster exports the following actions:
    * fetchClusters: 3 occurrences in sagas/clusterSagas, sagas/integrationSagas, sagas/networkSagas
    * fetchCluster: 1 occurrence in sagas/clusterSagas
    * selectCluster: 0 occurrences
    * startWizard: 0 occurrences
    * nextWizardPage: 0 occurrences
    * prevWizardPage: 0 occurrences
    * updateWizardState: 2 occurrences in sagas/clusterSagas
    * finishWizard: 0 occurrences
    * saveCluster: 2 occurrences in sagas/clusterSagas
    * deleteClusters: 1 occurrence in sagas/clusterSagas
    * downloadClusterYaml: 1 occurrence in sagas/clusterSagas
    * search actions: residue

3. sagas/clusterSagas watches the following locations to make /v1/clusters request:
    * /main/dashboard
    * /main/integrations
    * /main/policies
    * /main/compliance

4. sagas/integrationSagas refers to `fetchClusters` action, but `'clusters'` is not a source from which to delete integrations

5. sagas/networkSagas watches the following location to make /v1/clusters request:
    * /main/network

**Conclusion**: Network is the only container which gets clusters from redux and sagas.

### Changed files

1. Edit sagas/clusterSagas.js
    * Delete `watchLocation` function

2. Delete sagas/clusterSagas.test.js

3. Edit sagas/integrationSagas.js
    * Delete reference to `fetchClusters` action

### Residue

1. Delete more leftovers from reducers/clusters.js

## Checklist
- [x] Deleted unit tests

## Testing Performed

1. Visit paths to see /v1/clusters request with `watchLocation` and then no request without `watchLocation`
    * /main/dashboard
    * /main/integrations
    * /main/policies
    * /main/compliance

2. Visit path to see /v1/clusters request both with `watchLocation` without `watchLocation`
    * /main/network
